modules/server/src/main/scala/navigate/server/NavigateEngine.scala

```diff
@@ -410,4 +464,65 @@ object NavigateEngine {
     }
   )
 
+  private def command[F[_]: MonadThrow: Logger](
```

Comment:
Similar to `simple command`, but here the command is a `Handler`, executed in the state machine. This gives the command access to the global state, and allows it to have several stages.